pandas/_libs/lib.pyx

```diff
@@ -1698,6 +1698,48 @@ cpdef bint is_string_array(ndarray values, bint skipna=False):
     return validator.validate(values)
 
 
+cpdef ndarray ensure_string_array(
```

Comment:
this looks like a generalization of: https://github.com/pandas-dev/pandas/blob/master/pandas/_libs/lib.pyx#L621

pls de-dupe these as appropriate

Reply:
astype_str was only used in once, so I`ve deleted it and use ensure_string_array instead.